content/blog/rails-dynamic-translation-constants.md

```diff
@@ -0,0 +1,156 @@
++++
+title = "Dynamic Translations in Rails: Why Constants Can’t Adapt to Locale Changes and How to Fix It?"
+slug = "rails-dynamic-translation-constants"
+date = 2024-11-04T10:35:08+05:30
+image = "/images/2023/ruby-generic.png"
+draft = true
+authors = ["Nitin Rajkumar"]
+description = ""
+tags = ["Playbook", "Software Craftsmanship", "Rails", "I18n", "Translations Not Reflecting", "Numeronyms"]
+categories = ["Playbook", "Software Craftsmanship"]
+type = ""
++++
+
+In math and science a particular number or value that does not change is called a constant. In programming too, a constant typically represents a value that doesn’t change — a number, string, or any data type that remains the same throughout the application’s runtime. But what if you need to assign a dynamic, changing value to a constant? Can a constant, by its nature, ever be dynamic?
+
+Let’s explore this concept using a practical example in Rails and learn why, when working with multilingual support (I18n), a constant might not behave as expected.
+
+---
+
+### Setting Up a Constant for a Static Value
+
+Imagine you want to return a plain string whose value will never change upon request. A constant might seem like a good choice:
+
+```ruby
+class ReturnPlainString
+  PLAIN_STRING = {
+    title: "Meaningless title ever",
+    subtitle: "It's hard to explain you"
+  }
+
+  def simple_request
+    "1. " + PLAIN_STRING[:title]
+  end
+end
+
+ReturnPlainString.new.simple_request
+#=> "1. Meaningless title ever"
+
+```
+
+Everything works fine until now, not only the people from your country but people from other countries want to make use of this wonderful application you have built. You just can't ask them to go ahead and use it because they don't understand your language. So you decide to make you app available in all the languages.
```

Comment:
Felt like a break in the flow, suggestion:

Everything works fine until now, but what about people from other countries who want to use your application?
... Continued